<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Graphe Image &amp; Algorithme de Dijkstra ‚ú®</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
QMainWindow, QWidget#centralwidget {
    background-color: #3b2857;
    color: #ffffff;
}
QGroupBox {
    background-color: #513e71;
    border-radius: 8px;
    margin-top: 10px;
    font-weight: bold;
    color: #ffd700;
}
QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center;
    padding: 0 5px;
}
QToolButton, QPushButton {
    padding: 8px 15px;
    border-radius: 5px;
    font-weight: bold;
    border: none;
    color: white;
    min-height: 25px;
}
QToolButton#loadButton { background-color: #7b4397; }
QToolButton#resetButton { background-color: #ff9933; color: black; }

/* --- STYLE DES BOUTONS DE CONNEXIT√â --- */
/* √âtat par d√©faut (Non s√©lectionn√©) */
QToolButton#size16Button, QToolButton#size32Button { 
    background-color: #453a66; 
    border: 1px solid #7b4397; 
    color: #cccccc;
}
/* √âtat S√âLECTIONN√â (Checked) - C'est ici que la couleur change */
QToolButton#size16Button:checked, QToolButton#size32Button:checked {
    background-color: #9c27b0; /* Violet Vif */
    border: 2px solid #ffd700; /* Bordure Jaune */
    color: #ffffff;
}
/* √âtat Survol */
QToolButton#size16Button:hover, QToolButton#size32Button:hover {
    background-color: #6a1b9a;
}

QPushButton#calculateButton { background-color: #4CAF50; font-size: 12pt; padding: 12px; }
QPushButton#calculateButton:disabled { background-color: #555; color: #aaa; }
QScrollArea { border: 2px solid #7b4397; border-radius: 8px; background-color: #2a2a4a; }
QLabel#imageDisplayLabel { background-color: #2a2a4a; }
QLabel { color: #eeeeee; }
QLabel#dimValueLabel, QLabel#nodesValueLabel, QLabel#pathLengthValueLabel, QLabel#totalCostValueLabel, QLabel#connValueLabel { font-weight: bold; color: #ffffff; }
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainVLayout">
    
    <item>
     <widget class="QLabel" name="mainTitle">
      <property name="text"><string>Graphe Image &amp; Algorithme de Dijkstra ‚ú®</string></property>
      <property name="alignment"><set>Qt::AlignmentFlag::AlignCenter</set></property>
      <property name="font"><font><pointsize>16</pointsize><weight>75</weight><bold>true</bold></font></property>
     </widget>
    </item>

    <item>
     <widget class="QGroupBox" name="controlsBox">
      <property name="title"><string>Contr√¥les</string></property>
      <layout class="QGridLayout" name="controlsGridLayout">
       <item row="0" column="0"><widget class="QToolButton" name="loadButton"><property name="text"><string>‚¨ÜÔ∏è Charger Image</string></property><property name="sizePolicy"><sizepolicy hsizetype="Expanding" vsizetype="Fixed"><horstretch>0</horstretch><verstretch>0</verstretch></sizepolicy></property></widget></item>
       <item row="0" column="1"><widget class="QToolButton" name="resetButton"><property name="text"><string>üîÅ R√©initialiser</string></property><property name="sizePolicy"><sizepolicy hsizetype="Expanding" vsizetype="Fixed"><horstretch>0</horstretch><verstretch>0</verstretch></sizepolicy></property></widget></item>
       <item row="0" column="2">
        <layout class="QHBoxLayout" name="zoomLayout">
         <item><widget class="QLabel" name="zoomLabel"><property name="text"><string>Zoom:</string></property></widget></item>
         <item><widget class="QSlider" name="zoomSlider"><property name="orientation"><enum>Qt::Orientation::Horizontal</enum></property><property name="minimum"><number>10</number></property><property name="maximum"><number>200</number></property><property name="value"><number>100</number></property></widget></item>
         <item><widget class="QLabel" name="zoomValueLabel"><property name="text"><string>100%</string></property></widget></item>
        </layout>
       </item>
       
       <item row="1" column="0">
        <widget class="QToolButton" name="size16Button">
         <property name="text"><string>4-Connexit√© (H/B/G/D)</string></property>
         <property name="checkable"><bool>true</bool></property>
         <property name="checked"><bool>true</bool></property>
         <property name="sizePolicy"><sizepolicy hsizetype="Expanding" vsizetype="Fixed"><horstretch>0</horstretch><verstretch>0</verstretch></sizepolicy></property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QToolButton" name="size32Button">
         <property name="text"><string>8-Connexit√© (Diagonales)</string></property>
         <property name="checkable"><bool>true</bool></property>
         <property name="sizePolicy"><sizepolicy hsizetype="Expanding" vsizetype="Fixed"><horstretch>0</horstretch><verstretch>0</verstretch></sizepolicy></property>
        </widget>
       </item>
       
       </layout>
     </widget>
    </item>

    <item>
     <widget class="QScrollArea" name="imageScrollArea">
      <property name="sizePolicy"><sizepolicy hsizetype="Expanding" vsizetype="Expanding"><horstretch>0</horstretch><verstretch>10</verstretch></sizepolicy></property>
      <property name="minimumSize"><size><width>0</width><height>300</height></size></property>
      <property name="widgetResizable"><bool>true</bool></property>
      <property name="alignment"><set>Qt::AlignmentFlag::AlignCenter</set></property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <layout class="QVBoxLayout" name="scrollLayout">
        <item><widget class="QLabel" name="imageDisplayLabel"><property name="alignment"><set>Qt::AlignmentFlag::AlignCenter</set></property><property name="text"><string>Aucune image charg√©e. Utilisez le bouton ci-dessus.</string></property></widget></item>
       </layout>
      </widget>
     </widget>
    </item>

    <item>
     <layout class="QHBoxLayout" name="bottomLayout">
      <item>
       <layout class="QVBoxLayout" name="leftInfoLayout">
        <item>
         <widget class="QGroupBox" name="instructionsBox">
          <property name="title"><string>Instructions</string></property>
          <layout class="QVBoxLayout">
           <item><widget class="QLabel" name="instr1"><property name="text"><string>1. Clic Gauche : Point de D√©part (Bleu)</string></property></widget></item>
           <item><widget class="QLabel" name="instr2"><property name="text"><string>2. Clic Gauche : Point d'Arriv√©e (Rouge)</string></property></widget></item>
           <item><widget class="QLabel" name="instr3"><property name="text"><string>3. Cliquez sur &quot;Calculer&quot; ou changez de mode.</string></property></widget></item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="pointsBox">
          <property name="title"><string>Points S√©lectionn√©s</string></property>
          <layout class="QFormLayout">
           <item row="0" column="0"><widget class="QLabel" name="lblStart"><property name="text"><string>D√©part:</string></property></widget></item>
           <item row="0" column="1"><widget class="QLabel" name="startPointLabel"><property name="text"><string>N/A</string></property><property name="styleSheet"><string>color: #55aaff; font-weight: bold;</string></property></widget></item>
           <item row="1" column="0"><widget class="QLabel" name="lblEnd"><property name="text"><string>Arriv√©e:</string></property></widget></item>
           <item row="1" column="1"><widget class="QLabel" name="endPointLabel"><property name="text"><string>N/A</string></property><property name="styleSheet"><string>color: #ff5555; font-weight: bold;</string></property></widget></item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>

      <item>
       <widget class="QGroupBox" name="resultBox">
        <property name="title"><string>Analyse &amp; R√©sultats</string></property>
        <layout class="QGridLayout" name="resultGridLayout">
         <item row="0" column="0"><widget class="QLabel" name="l1"><property name="text"><string>Dimensions:</string></property></widget></item>
         <item row="0" column="1"><widget class="QLabel" name="dimValueLabel"><property name="text"><string>N/A</string></property></widget></item>
         <item row="1" column="0"><widget class="QLabel" name="l2"><property name="text"><string>Sommets:</string></property></widget></item>
         <item row="1" column="1"><widget class="QLabel" name="nodesValueLabel"><property name="text"><string>N/A</string></property></widget></item>
         <item row="2" column="0"><widget class="QLabel" name="l3"><property name="text"><string>Connexit√©:</string></property></widget></item>
         <item row="2" column="1"><widget class="QLabel" name="connValueLabel"><property name="text"><string>4-connexit√©</string></property></widget></item>
         <item row="3" column="0"><widget class="QLabel" name="l4"><property name="text"><string>Longueur:</string></property></widget></item>
         <item row="3" column="1"><widget class="QLabel" name="pathLengthValueLabel"><property name="text"><string>0</string></property></widget></item>
         <item row="4" column="0"><widget class="QLabel" name="l5"><property name="text"><string>Co√ªt Total:</string></property></widget></item>
         <item row="4" column="1"><widget class="QLabel" name="totalCostValueLabel"><property name="text"><string>0.00</string></property></widget></item>
         <item row="5" column="0"><widget class="QLabel" name="l6"><property name="text"><string>Explor√©s:</string></property></widget></item>
         <item row="5" column="1"><widget class="QLabel" name="visitedNodesValueLabel"><property name="text"><string>0</string></property></widget></item>
         <item row="6" column="0" colspan="2"><widget class="QLabel" name="resultStatusLabel"><property name="text"><string>En attente...</string></property><property name="alignment"><set>Qt::AlignCenter</set></property><property name="styleSheet"><string>background-color: #2a2a4a; padding: 5px; border-radius: 4px;</string></property></widget></item>
         <item row="7" column="0" colspan="2"><widget class="QPushButton" name="calculateButton"><property name="enabled"><bool>false</bool></property><property name="text"><string>‚ö° Calculer le chemin</string></property><property name="cursor"><cursorShape>PointingHandCursor</cursorShape></property></widget></item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>